### PR TITLE
Converted cutmix example to Keras Core

### PR DESCRIPTION
**Pull request by cosmo3769**
Converted the keras-io example [CutMix data augmentation for image classification](https://keras.io/examples/vision/cutmix/) to `keras-core`.

[Colab Notebook link](https://colab.research.google.com/drive/1EwPk-5E38A5460RHbLOMo6HDZv5aOVwZ?usp=sharing)

### CODE REVIEW

File: examples/keras_io/tensorflow/vision/cutmix.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+"""
+## Setup
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import tensorflow as tf
+from tensorflow import keras
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/cutmix.py:52]: There should be no `tf.keras` mention and `keras` should be imported as `import keras_core as keras`.

Comment (cosmo3769) [reply examples/keras_io/tensorflow/vision/cutmix.py]: Yeah, I didn't pushed the changes then. 😅

Comment (cosmo3769) [reply examples/keras_io/tensorflow/vision/cutmix.py]: Now, I have made the necessary changes and also attached the colab link in the description above to show that it is working correctly.

File: examples/keras_io/tensorflow/vision/cutmix.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import tensorflow as tf
+from keras_core import layers
+import keras_core as keras
+
+np.random.seed(42)
+tf.random.set_seed(42)
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/cutmix.py:57]: You can use `keras.utils.set_random_seed(42)` to replace these 2 lines

File: examples/keras_io/tensorflow/vision/cutmix.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+"""
+## Notes
+
+In this example, we trained our model for 15 epochs.
+In our experiment, the model with CutMix achieves a better accuracy on the CIFAR-10 dataset
+(76.92% in our experiment) compared to the model that doesn't use the augmentation (72.23%).
+You may notice it takes less time to train the model with the CutMix augmentation.
+
+You can experiment further with the CutMix technique by following the
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/cutmix.py:378]: You can remove the hugging face paragraph since it isn't relevant to the code anymore

**Review by fchollet — COMMENTED** [review]
Thanks for the update! Can you post the git diff from the original example?

**Comment by cosmo3769** [comment]
> Can you post the git diff from the original example?

Here is the patch file showing git diff: [my_changes.patch](https://github.com/keras-team/keras-core/files/12139156/my_changes.patch). Thank you.

**Review by fchollet — APPROVED** [review]
LGTM -- thank you for the contribution!

**Pull request merged**